issue#1 return Json file

417-develop opened the pull request:
- add static folder
- add GET json endpoint

Related 用意したJSONファイルを返すようにしてほしい #1 
Related エンドポイントごとに返すJSONファイルを変えたい #2 
Related 指定したリソースごとにJSONファイルを変えたい #3

417-develop committed:
issue#1 return Json file

- add static folder
- add GET json endpoint

417-develop committed:
issue#3 return multiple json in one ep

- Add ner endpoint which has a dynamic parameter
- Add car folder

j123 commented on main.go:
```diff
@@ -1,16 +1,44 @@
+var router *gin.Engine
+
+func main() {
+	router = gin.Default()
```
VS CodeってインデントがTABになるんだね。IntelliJみたいにSpaceにできないのかな？
あとでVS Codeの設定を調べてみる。

417-develop replied:
右下のにこちゃんが書いてあるステータスバーに `Space: 4` って書いてありまして、
そちらをクリックして、`Convert indention to Spaces` を見つけました！

j123 replied:
Great!!! 👍

j123 commented on main.go:
```diff
@@ -1,16 +1,44 @@
+}
+
+func getFoo(c *gin.Context) {
+	bytes, e := ioutil.ReadFile(`./static/json/get-foo.json`)
```
なんか、こんな書き方が普通らしいぞ。
```
  if bytes, e := ioutil.ReadFile(`./static/json/get-foo.json`); e == nil {
    data, _ := simplejson.NewJson(bytes)
    fmt.Println(data)
    c.JSON(http.StatusOK, data)
  } else {
    log.Fatal(e)
  }
```

j123 replied:
(j123 はこの書き方で書いてないけど)

417-develop replied:
エラーになる条件がわかりやすいので、if文でエラーは先に排除するのが、個人的には好みですが、
goのエラーキャッチはこれがスマートなのですね、、、！！！
確かに引数として帰ってくるし、1コマンドで終わらせようぜって考えなのですかね。
ありがとうございます、修正致します。

417-develop replied:
（そしてすっかり `_` の存在忘れてました、、、Thank you sooo much)

j123 replied:
👍 スッキリしたね

j123 replied:
pullって実行してみたら、
```
if bytes, e := ioutil.ReadFile(`./static/json/get-foo.json`); e == nil {
    (省略)
} else {
    c.JSON(http.StatusNotFound, gin.H{"message": "Server Error"})
}
```
だとうまくいかなかった。
(elseの中には入るけど、 `c.JSON()` が返されず、goが落ちる)

なので👇のように元に戻したら、ちゃんと `c.JSON()` を返すようになった。
```
    bytes, e := ioutil.ReadFile(`./static/json/car/get-car` + id + `.json`)
    if e != nil {
        c.JSON(http.StatusNotFound, gin.H{"message": "Server Error"})
    }
    data, _ := simplejson.NewJson(bytes)
　　(省略)
```

うーん。
ということで、スミマセン。元の方式で:bow:

j123 replied:
あ、ついでに、 `func getCar()` だけは、 `http.StatusInternalServerError` じゃなくて `http.StatusNotFound` の方がAPIっぽくて（該当IDは存在しない）良いと思います！

417-develop replied:
<img width="1084" alt="screen shot 2018-12-12 at 0 04 10" src="https://user-images.githubusercontent.com/25281178/49809138-7cca6100-fda1-11e8-94a2-4473eb293e99.png">
私のローカルでは動きました 🤔 (何か勘違いしていたらすみません）

j123 replied:
むむ。動いてるね。

> 何か勘違いしていたらすみません

こちらの勘違いとみた。
なので問題なし。

j123 replied:
> これ入れましたが、 os.Exit(1). でシステム終了させちゃうみたいです、、消します

`git pull` をするのを忘れてて最新ではなく `log.Fatal(e)` が残ってる状態のを実行してたからだった。

417 san の全面勝訴です:bow:

j123 commented on main.go:
```diff
@@ -1,16 +1,44 @@
+func getFoo(c *gin.Context) {
+	bytes, e := ioutil.ReadFile(`./static/json/get-foo.json`)
+	if e != nil {
+		log.Fatal(e)
```
エラーの場合ここでリターンしちゃわないと、後続の処理でエラっちゃうね
```
c.JSON(http.StatusInternalServerError, gin.H{"message": "Server Error"}
return
```
みたいな感じで。

j123 replied:
あ、 `returmn` は要らないか(も)

417-develop replied:
ありがとうございます！
（御察しの通り、修正後は後続の処理がなくなったので、return入れませんでした。）

j123 commented on main.go:
```diff
@@ -1,16 +1,44 @@
+	if e != nil {
+		log.Fatal(e)
+	}
+	data, e := simplejson.NewJson(bytes)
```
エラーの場合ここでリターン(ry

j123 reviewed (commented):
94点:smile:

417-develop committed:
Merge pull request #5 from 417-develop/3_multi_json

issue#3 return multiple json in one ep

417-develop commented on main.go:
```diff
@@ -1,16 +1,56 @@
+		log.Fatal(e)
+	}
+	data, e := simplejson.NewJson(bytes)
+	fmt.Println(data)
```
👿

417-develop committed:
Modify methods code style

- convert tab to space
- chenge the branch line
- add returning the 500 when there is an error

417-develop commented on main.go:
```diff
@@ -1,16 +1,55 @@
+        data, _ := simplejson.NewJson(bytes)
+        c.JSON(http.StatusOK, data)
+    } else {
+        log.Fatal(e)
```
これ入れましたが、 `os.Exit(1).` でシステム終了させちゃうみたいです、、消します 😨 
https://golang.org/pkg/log/#Fatal

j123 replied:
> os.Exit(1). でシステム終了

最強コマンドｗ

417-develop committed:
remove unnecessary loggers

417-develop committed:
Changed the error response of GET /car

j123 commented:
LGTMです！！
